src/jdk.jpackage/share/classes/jdk/jpackage/internal/model/Application.java

```diff
@@ -88,6 +88,17 @@ public interface Application extends BundleSpec {
      */
     Optional<Path> srcDir();
 
+    /**
+     * Gets the source directory of this application if available or an empty
+     * {@link Optional} instance.
+     * <p>
+     * Source directory is a directory with the applications's classes and other
+     * resources.
+     *
+     * @return the source directory of this application
+     */
+    Optional<Path> runtimeImage();
+
```

Comment:
What is "the source directory of this application"?
Why is the method that returns a path to the source directory called `runtimeImage()`?

Reply:
> runtimeImageDir is a value of --runtime-image in case of runtime installer.

This duplicates `Package.predefinedAppImage()` function. In case of runtime packaging, the "predefined app image" should be the value of `--runtime-image`. Can you evaluate if you can use `Package.predefinedAppImage()` function instead of adding a new one?